test/Microsoft.ML.Tests/OnnxTests.cs

```diff
@@ -51,6 +53,72 @@ public class BreastCancerMCPrediction
             public float[] Scores;
         }
 
+        [Fact]
+        public void InitializerCreationTest()
+        {
+            using (var env = new ConsoleEnvironment())
+            {
+                // Create the actual implementation
+                var ctxImpl = new OnnxContextImpl(env, "model", "ML.NET", "0", 0, "com.test");
+
+                // Use implementation as in the actual conversion code
+                var ctx = ctxImpl as OnnxContext;
+                ctx.AddInitializer(9.4f, "float");
```

Comment:
>ctx.AddInitializer(9.4f, "float"); [](start = 16, length = 34)

var floatScalar = ctx.AddInitializer(9.4f, "float");

Reply:
Why do we need this?

Reply:
it's better than do 
```var floatScalar = model.Graph.Initializer[0];```

If someone decide to change this test, and add one more initializer in the beginning, it will break tests.

---
In reply to: [220259227](https://github.com/dotnet/machinelearning/pull/965#discussion_r220259227) [](ancestors = 220259227)

Reply:
ctx.AddInitializer returns the ONNX name of a tensor, not the actual ONNX object.

Reply:
Can you use that name of tensor to access initializer instead of model.Graph.Initalizer[0]? 
If yes, i would suggest to use names instead of index.
If no, I find it strange, but i'm ok with current code.

---
In reply to: [220418126](https://github.com/dotnet/machinelearning/pull/965#discussion_r220418126) [](ancestors = 220418126)